### PR TITLE
Improve CI build

### PR DESCRIPTION
**Pull request by rst0git**
This pull request replaces the hard-coded number of make jobs (`-j2`) with a number corresponding to the processing units available. This allows `docker build` to complete faster on systems with more than 2 CPU cores. In addition, the verbosity of `apt-get` has been reduced to avoid logging messages such as download progress in Travis CI.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,11 +2,6 @@ ARG PARENT_VERSION=latest
 FROM p4lang/pi:${PARENT_VERSION}
 LABEL maintainer="Antonin Bas <antonin@barefootnetworks.com>"
 
-# Default to using 2 make jobs, which is a good default for CI. If you're
-# building locally or you know there are more cores available, you may want to
-# override this.
-ARG MAKEFLAGS=-j2
```
Comment (antoninbas) [inline Dockerfile:8]: do you think you could update https://github.com/p4lang/behavioral-model/blob/main/Dockerfile.noPI as well?

File: .travis.yml [hunk 2]
```diff
@@ -47,6 +47,6 @@ install:
 
 script:
   - ci_env=`bash <(curl -s https://codecov.io/env)`
-  - docker run $ci_env --env GCOV -w /behavioral-model bm /bin/bash -c "make check -j2 && ./travis/codecov.sh"
-  - if [ "$sswitch_grpc" = "yes" ]; then docker run -w /behavioral-model/targets/simple_switch_grpc bm make check -j2; fi
+  - docker run --rm $ci_env --env GCOV -w /behavioral-model bm /bin/bash -c "make check -j$(nproc) && ./travis/codecov.sh"
+  - if [ "$sswitch_grpc" = "yes" ]; then docker run --rm -w /behavioral-model/targets/simple_switch_grpc bm make check -j$(nproc); fi
```
Comment (antoninbas) [inline .travis.yml:51]: well the CI VMs are not reused, so I would say `--rm` doesn't add value, but there is probably no harm in adding it either

File: Dockerfile (hunk 1 above)
Comment (rst0git) [reply Dockerfile:8]: done

File: .travis.yml (hunk 2 above)
Comment (rst0git) [reply .travis.yml:51]: I agree, it is helpful when copy/paste the docker command from CI logs to replicate an error.

**Review by antoninbas — APPROVED** [review]

**Pull request merged**